lib/m2ee/munin.py

```diff
@@ -174,55 +217,185 @@ def get_stats_from_runtime(client, config):
     m2eeresponse = client.server_statistics()
     if not m2eeresponse.has_error():
         stats.update(m2eeresponse.get_feedback())
-    if type(stats['requests']) == list:
+    if type(stats["requests"]) == list:
         # convert back to normal, whraagh
         bork = {}
-        for x in stats['requests']:
-            bork[x['name']] = x['value']
-        stats['requests'] = bork
+        for x in stats["requests"]:
+            bork[x["name"]] = x["value"]
+        stats["requests"] = bork
 
     runtime_version = config.get_runtime_version()
     if runtime_version is not None and runtime_version >= 3.2:
         m2eeresponse = client.get_all_thread_stack_traces()
         if not m2eeresponse.has_error():
-            stats['threads'] = len(m2eeresponse.get_feedback())
+            stats["threads"] = len(m2eeresponse.get_feedback())
 
     java_version = guess_java_version(client, runtime_version, stats)
-    if 'memorypools' in stats['memory']:
-        memorypools = stats['memory']['memorypools']
+    return _populate_stats_by_java_version(stats, java_version), java_version
+
+
+def _populate_stats_by_java_version(stats, java_version):
+    # Notes from writing this function:
+    # Explanation of what each of these graphs mean:
+    # https://github.com/mendix/m2ee-tools/blob/master/doc/munin.md#the-smaps-statistics
+
+    # Explanation with some visuals (expand this comment)
+    # https://stackoverflow.com/questions/1262328/how-is-the-java-memory-pool-divided
+
+    # A memorypool is defined by the Jave docs as the following:
+    # "A memory pool represents the memory resource managed by the Java virtual
+    # machine and is managed by one or more memory managers."
+    # (source: https://docs.oracle.com/en/java/javase/11/docs/api/java.management/java/lang/management/MemoryPoolMXBean.html)  pylint: disable=line-too-long, noqa: E501
+
+    # The Mendix runtime uses the standard MemoryMXBean:
+    # The values from the `MemoryPoolMXBean`s go into the memorypool field and
+    # the values from the `MemoryMXBean` go into the init_, committed_, used_
+    # and max_ fields, both for the _heap and the nonheap memory.
+
+    # Since apparently in different Java versions these mean different things
+    # or are in different places, we need to shuffle these around depending
+    # on the Java version.
+
+    if "memorypools" in stats["memory"]:
+        standardized_memory_pools = _standardize_memory_pools_output(
+            stats["memory"]["memorypools"], java_version
+        )
+        stats["memory"].update(standardized_memory_pools)
+    elif java_version >= 8:
+        # We reach this branch only when we don't have the memorypools
```

Comment:
This comment is not accurate.

"When we don't have the memory pools added, which is for Mendix runtime 6 and Java 8" -> this statement is contradicted by the unit tests here: https://github.com/mendix/cf-mendix-buildpack/pull/320/files#diff-d81c0bb7df07ceb274c29675fb9060b0R44

I think that the comment [here](https://github.com/mendix/cf-mendix-buildpack/pull/320/files#diff-6d7767a80a7fe16cde455e3fd7e10638R371) is somewhat clearer, can we use part of that?

Reply:
I've updated the comment.